packages/dag4-wallet/src/shared/operations.ts

```diff
@@ -0,0 +1,165 @@
+import { keyStore } from "@stardust-collective/dag4-keystore";
+import {
+  AllowSpendWithCurrencyId,
+  DagNetwork,
+  GlobalDagNetwork,
+  HashResponse,
+  MetagraphTokenNetwork,
+  SignedAllowSpend,
+  SignedTokenLock,
+  TokenLockWithCurrencyId,
+  TokenLockWithParent,
+  TransactionReference,
+} from "@stardust-collective/dag4-network";
+import { normalizePublicKey } from "../utils";
+import {
+  dagAddressValidator,
+  allowSpendSchema,
+  tokenLockSchema,
+  validateArraySchema,
+  validateSchema,
+} from "../validationSchemas";
+import { DagAccount } from "../dag-account";
+
+type SharedNetwork = DagNetwork | GlobalDagNetwork | MetagraphTokenNetwork;
+
+export const allowSpend = async (
+  body: AllowSpendWithCurrencyId,
+  network: SharedNetwork,
+  account: DagAccount
```

Comment:
You should be careful introducing complex dependencies in a function like this. Is there any reason we need a DagAccount object here? We just need address and keyTrio. 

Any assertions on the account object should be done in the account class imo.